### ISSUE TITLE
[FtrRequest] Add a possibility to handle broadcasted and message publishing from embedding application

### ISSUE THREAD

**Issue by johnsirach** [moquette-io/moquette#113]
As requested from #110

Too enhance embedding it would be nice to have a mechanism which is a able to supply local listeners to handle broadcast messages to the moquette broker and a mechanism to inject messages to the broker which the broker broadcasts. 

This will make it possible to still handle MQTT messages without the need to have both the moquette broker and a client running at the same time in case the embedding application needs the messages.

A Listener interface could have (abstract):
`handleFromBroker(String clientId, String topic, ByteBuffer message);`

Code for supplying messages to the broker could be like (abstract):

```
PublishMessage message = new PublishMessage();
message.setTopicName("/path/supplied/to/publish");
message.setRetainFlag(false);
message.setQos(AbstractMessage.QOSType.MOST_ONE);
message.setPayload(ByteBuffer.wrap("sendvalue".getBytes()));
moquette->getProcessor().processInternalPublish(message); 
```

Mentioned in #110 is to use Reactor or other system to handle the events to the moquette embedding host/code

**Comment by andsel** [comment]
Hi @johnsirach fixed with commit #16a5556. To publish you use the `Server.internalPublish(PublishMessage msg);` to listen instead look at EmbeddedLauncher example, essentially you should register your implementation of AbstractInterceptHandler.

 Andrea

**Comment by johnsirach** [comment]
Awesome for adding this! The other two requests are less critical for my current setup so will be able to test this asap. I will reply back if requested after getting the results.
